docs/Planner.md

```diff
@@ -0,0 +1,108 @@
+## AST matching mechanics
+
+### Example 1
+
+Here is a query graph (green theme) and an index scan graph (red theme). We would like to match the query graph against that index graph. In this case the index scan graph uses an outside parameter that needs to be filled in as a byproduct of the matching. The second byproduct is the necessary compensation of the match. The query graph is independent of any outside parameters.
```

Comment:
This is perhaps a bigger change as it may require redoing the graphs, but red and green may not be the greatest choices as they can be difficult to distinguish for people with red-green colorblindness (the most common kind of colorblindness). It may also be sufficient to just relabel them as "green theme on the left" and "red theme on the right".